### ISSUE TITLE
Quick Fix import puts local imports with third-party imports

### ISSUE THREAD

**Issue by euan-reid** [microsoft/pylance-release#4872]
## Environment data

-   Language Server version: 2023.9.20
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.11.5
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: strict

Using poetry 1.6.1, python and packages installed in venv

## Code Snippet

![import-order](https://github.com/microsoft/pylance-release/assets/1822409/f2cd63a0-4b62-4ec1-aa70-7fa32988a0eb)

## Expected behavior

Quick Fix-injected local imports are grouped together after third party imports

## Actual behavior

Quick Fix-injected local imports are mixed in with third party imports

## Logs

```
2023-09-23 13:59:37.877 [info] (87036) [BG(1)] analyzing: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py ...
2023-09-23 13:59:37.877 [info] (87036) [BG(1)]   parsing: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (2ms)
2023-09-23 13:59:37.877 [info] (87036) [BG(1)]   binding: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (1ms)
2023-09-23 13:59:37.897 [info] (87036) [BG(1)]   checking: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (19ms)
2023-09-23 13:59:37.897 [info] (87036) [BG(1)] analyzing: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (22ms)
2023-09-23 13:59:37.898 [info] (87036) [BG(1)] indexing: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py [found 1] (0ms)
2023-09-23 13:59:37.907 [info] (87036) [FG] parsing: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (1ms)
2023-09-23 13:59:37.907 [info] (87036) [FG] binding: /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (0ms)
2023-09-23 13:59:38.069 [info] (87036) [BG(1)] getSemanticTokens delta previousResultId:1695502767600 at /Users/euan/git/masters-tavern/masters_tavern/models/identity.py (1ms)
```

**Comment by rchiodo** [comment]
Thanks for the issue. Can you share your code or an example that reproduces the problem? With a trivial repro, I can't seem to reproduce:

![autoimport](https://github.com/microsoft/pylance-release/assets/19672699/2c6165c9-2278-45e7-a315-dca1ada07552)

My guess is that we think masters_tavern is a 3rd party import for some reason.

**Comment by euan-reid** [comment]
Sorry for the delay in responding. I've made a smallest repro for the behaviour [here](https://github.com/euan-reid/pyright-bug-repro).

> My guess is that we think masters_tavern is a 3rd party import for some reason.

That's my suspicion too, a lot of my initial searching was in the vein of `pylance local "third party" import wrong`. I thought it might be the re-exporting in some of the `__init__.py` files, but with the minimum repro having all `__init__.py` files empty it seems like I was wrong there.

In case it helps, for additional context I'm running on a 2021 Macbook Pro with an M1 Pro chip with OS version Ventura 13.5.2.

**Comment by rchiodo** [comment]
Thanks, that example lets me reproduce the problem.

I don't think this is very high priority at the moment though, so it will probably be a while before we get to fixing it.

**Comment by bschnurr** [comment]
This issue has been fixed in prerelease version 2024.2.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20242100-13-February-2024-PreRelease)
